client/src/main/scala/com.crobox.clickhouse/internal/ClickHouseExecutor.scala

```diff
@@ -41,12 +94,15 @@ private[clickhouse] trait ClickHouseExecutor extends LazyLogging {
   def executeRequest(query: String,
                      settings: QuerySettings,
                      entity: Option[RequestEntity] = None,
-                     progressQueue: Option[SourceQueueWithComplete[QueryProgress]] = None): Future[String] =
+                     progressQueue: Option[SourceQueueWithComplete[QueryProgress]] = None): Future[String] = {
+    val queryIdentifier = Random.alphanumeric.take(20).mkString("")
```

Comment:
Is this the kind of identifier we want? is this proper randomness?

Reply:
This is just an internal identifier for now so it should be perfectly fine. We might want to reconsider when we'll expose it to the outside